modules/web/js/ballerina/components/return-parameter-definition-holder.jsx

```diff
@@ -52,30 +53,28 @@ class ReturnParameterDefinitionHolder extends React.Component {
      * @return {boolean} true||false
      * */
     addReturnParameter(input) {
+        const fragment = FragmentUtils.createReturnParameterFragment(input);
+        const parsedJson = FragmentUtils.parseFragment(fragment);
         const model = this.props.model;
-        const splitedExpression = input.split(' ');
 
-        const parameterDef = model.getFactory().createParameterDefinition();
-        const bType = splitedExpression[0];
-        if (this.validateType(bType)) {
-            parameterDef.setTypeName(bType);
+        if ((!_.has(parsedJson, 'error') && !_.has(parsedJson, 'syntax_errors'))) {
+            if (_.isEqual(parsedJson.type, 'parameter_definition')) {
+                let parameterDefinition = model.getFactory().createParameterDefinition(parsedJson);
+                parameterDefinition.initFromJson(parsedJson);
+                if (!this.checkWhetherIdentifierAlreadyExist(parsedJson.parameter_name)) {
+                    this.props.model.addChild(parameterDefinition);
```

Comment:
This is where i only attach new node to the model. is doing this silently correct?